packages/venia-ui/lib/drivers/adapter.js

```diff
@@ -38,6 +39,65 @@ persistCache({
     storage: window.localStorage
 });
 
+const initialData = {
+    __typename: 'CartData',
+    gift_options: {
+        __typename: 'GiftOptions',
+        include_gift_receipt: false,
+        include_printed_card: false,
+        gift_message: ''
+    }
+};
+
+const GET_GIFT_OPTIONS_QUERY = gql`
+    query getGiftOptions {
+        gift_options @client {
+            __typename
+            include_gift_receipt
+            include_printed_card
+            gift_message
+        }
+    }
+`;
+
+const resolvers = {
+    Query: {
+        gift_options: (_, __, { cache }) => {
```

Comment:
I'm not sure if these can be async but you may consider moving the REST calls here to the resolvers. Basically you would only interact with graphql in the component/talon but the resolvers would actually handle doing the rest call and update the cache accordingly.

Reply:
Resolvers are temporary. I went with the easiest temporary option here. Making a network call from the hook gives me more control over when to make the call, for instance in case of gift message only send the request every 10 seconds. Yes, I can do this from the resolver as well, but the resolver can not differentiate between the checkbox and gift message.

Reply:
The flow I imagined was:

User input > call mutation > call rest > augment cache with rest results

Right now we're doing both sort of simultaneously, which isn't necessarily wrong I guess, but it puts all the logic of doing the fetch within the component in a manner that we're trying to move away from in the long run.

Reply:
@tjwiebell @jimbo lemme know if I'm trying to scope creep here.

Reply:
As of now we are only storing gift option choices and text to local storage through Apollo. When we work on Checkout we will need to make sure to add some step that pulls this data from the cache and submits it via REST when you click the checkout button.

Reply:
I would normally have done the API call on submit of the "step", which is `Proceed to Checkout` or some variation of `Continue`. With the way the forms are broken out, though, it may be easier to wait until `Place Order`.